### PR TITLE
Implement MGTransferGlobalCoarsening::memory_consumption()

### PR DESCRIPTION
**Pull request by peterrum**

### CODE REVIEW

File: include/deal.II/multigrid/mg_transfer_global_coarsening.h [hunk 1]
```diff
@@ -662,6 +683,23 @@ MGTransferGlobalCoarsening<dim, VectorType>::interpolate_to_mg(
...
+std::size_t
+MGTransferGlobalCoarsening<dim, VectorType>::memory_consumption() const
+{
+  std::size_t size = 0;
+
+  const unsigned int min_level = transfer.min_level();
+  const unsigned int max_level = transfer.max_level();
+
+  for (unsigned int l = max_level; l > min_level; --l)
+    size += this->transfer[l].memory_consumption();
```
Comment (bangerth) [inline include/deal.II/multigrid/mg_transfer_global_coarsening.h:698]: Not wrong though maybe uncommon to count down :-)

**Review by bangerth — COMMENTED** [review]
You declare three new functions, but implement only two. There must be a missing one :-)

File: include/deal.II/multigrid/mg_transfer_global_coarsening.h (hunk 1 above)
Comment (peterrum) [reply include/deal.II/multigrid/mg_transfer_global_coarsening.h:698]: > Not wrong though maybe uncommon to count down :-) 

I just copied the code from above.

> You declare three new functions, but implement only two. There must be a missing one :-)

This is fine as it is. `MGTwoLevelTransfer<dim, LinearAlgebra::distributed::Vector<Number>>` is the only class specialization of `MGTwoLevelTransfer`.

**Review by kronbichler — APPROVED** [review]

**Pull request merged**